src/dialogxml/widgets/button.cpp

```diff
@@ -164,13 +184,19 @@ void cButton::validatePostParse(ticpp::Element& elem, std::string fname, const s
 	if(labelledButtons.count(type)) {
 		if(!attrs.count("color") && !attrs.count("colour") && parent->getBg() == cDialog::BG_DARK)
 			setColour(sf::Color::White);
-		if(!getText().empty() && !attrs.count("width"))
+		if(type != BTN_TINY && type != BTN_LED && !getText().empty() && !attrs.count("width"))
 			throw xMissingAttr(elem.Value(), "width", elem.Row(), elem.Column(), fname);
 	}
 }
 
 location cButton::getPreferredSize() const {
-	return {btnRects[type][0].width(), btnRects[type][0].height()};
+	int width = btnRects[type][0].width();
+	if(type == BTN_TINY && !getText().empty()){
```

Comment:
Hmm… this could probably also be done for `BTN_PUSH` later (the label should probably be centered beneath the button in that case)… though I think it's unused right now so I guess it's not a priority.